evergreen.yml

```diff
@@ -118,6 +118,48 @@ functions:
           ${prepare_shell}
           cargo test
 
+  "install unix odbc":
+    - command: shell.exec
+      type: system
+      params:
+        shell: bash
+        working_dir: mongo-odbc-driver
+        script: |
+          ${prepare_shell}
+          dir="$PWD"
+          mkdir -p unixodbc/install
+          cd unixodbc
+          echo "downloading unixODBC"
+          unixODBC_dir=unixODBC-2.3.6
+          curl -O "http://noexpire.s3.amazonaws.com/sqlproxy/binary/linux/unixODBC-2.3.6.tar.gz" \
+            --silent \
+            --fail \
+            --max-time 60 \
+            --retry 5 \
+            --retry-delay 0
+          tar xf "$unixODBC_dir.tar.gz"
+          cd "$unixODBC_dir"
+          ./configure --prefix="$dir/unixodbc/install" --with-pic
+          make
+          make install
+
+
+  "run asan tests":
+    - command: shell.exec
+      type: test
+      params:
+        shell: bash
+        working_dir: mongo-odbc-driver
+        script: |
+          ${prepare_shell}
+          dir="$PWD"
+          ~/.cargo/bin/rustup default nightly
+          ~/.cargo/bin/rustup target add x86_64-unknown-linux-gnu
+          export LD_LIBRARY_PATH="$dir/unixodbc/install/lib"
+          export LIBRARY_PATH="$dir/unixodbc/install/lib"
+          export RUSTFLAGS="-Z sanitizer=address"
```

Comment:
What about leak and memory sanitizers? Can we run them too? Looking at the examples from rust-san readme, it seems that we could quite easily have leaks (for a newbie with the language like me at least).

Reply:
`address` actually catches `leaks`, I believe `leak` is just a faster version that catches only `leaks`. It seems faster and uses slightly different output for leaks, but I don't see a reason to run both when `address` catches leaks. What is `memory`, isn't that just `address`?

Reply:
oh, I was looking at the doc for the sanitizer here : https://github.com/japaric/rust-san
Seems like I should have look here : https://doc.rust-lang.org/beta/unstable-book/compiler-flags/sanitizer.html
Then, yes, address is enough :)